tests/Fixtures/Document/TestObject.php

```diff
@@ -0,0 +1,39 @@
+<?php
+/*
+ * Copyright 2023-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace MongoDB\Tests\Fixtures\Document;
+
+final class TestObject
+{
+    public int $id;
+
+    public TestNestedObject $x;
+
+    public bool $decoded;
+
+    public static function createForFixture(int $i, bool $decoded = false): self
+    {
+        $instance = new self();
+        $instance->id = $i;
+        $instance->decoded = $decoded;
+
+        $instance->x = new TestNestedObject();
+        $instance->x->foo = 'bar';
```

Comment:
I assume the `x` property and its value is not meant to be configurable, or you could always create a new factory method if needed?

Reply:
I could, but only some tests touch it at all. Since the properties are public, I decided to not add yet another factory or argument to change the value of `foo`.